qcodes_contrib_drivers/drivers/MontanaInstruments/cryostation.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+"""QCoDeS-Driver for Montana Instruments Cryostation.
+Tested on a cryostation s50
+https://www.montanainstruments.com/products/s50
+
+Authors:
+    Julien Barrier, <julien@julienbarrier.eu>
+"""
+
+import logging
+from time import sleep
+from typing import Optional, Any, Dict
+
+from qcodes.instrument import IPInstrument
+from qcodes.parameters import Parameter
+from qcodes import validators as vals
+
+log = logging.getLogger(__name__)
+
+
+class MontanaInstruments_Cryostation(IPInstrument):
```

Comment:
can't this instrument be connected to visa VISA? So the driver class inherits from VisaInstrument, and then for the adress you pass something like `TCPIP[board]::host address::port::SOCKET` (see here https://pyvisa.readthedocs.io/en/latest/introduction/names.html#visa-resource-syntax-and-examples )?

Reply:
IPInstrument is not recommended to be used, unless really needed, it's an old "hack" implementation in a way, so if it's possible to use VISA that'd be much more robust in terms of connectivity as well.

Reply:
Unfortunately this raises a VisaIOError. Trying to find the resource directly from pyvisa or NI Max is impossible.

Reply:
ok :(